### ISSUE TITLE
Symfony 4 support

### ISSUE THREAD

**Issue by slootjes** [knpuniversity/oauth2-client-bundle#65]
Is this compatible with Symfony 4? If not, what work needs to be done to make it work? If yes, can it be added to composer.json so it will pass the compatiblity checks?

**Comment by weaverryan** [comment]
Let's find out! :) #66

**Comment by slootjes** [comment]
Great, I'll try it tomorrow and will post my findings.

**Comment by weaverryan** [comment]
Just tagged 1.13.0 with Symfony 4 support - the tests all passed, so it should work fine. But if you find an issue, let us know!

**Comment by slootjes** [comment]
Awesome, thanks for your fast response :)

**Comment by slootjes** [comment]
I can not load 1.13 with composer in a v4.0.0-BETA2 project just yet:


Your requirements could not be resolved to an installable set of packages.

```
  Problem 1
    - Installation request for knpuniversity/oauth2-client-bundle ^1.13 -> satisfiable by knpuniversity/oauth2-client-bundle[1.13.0].
    - knpuniversity/oauth2-client-bundle 1.13.0 requires symfony/framework-bundle ^2.7|^3.0|4.0 -> satisfiable by symfony/framework-bundle[2.7.x-dev, 2.8.x-dev, 3.0.x-dev, 3.1.x-dev, 3.2.x-dev, 3.3.x-dev, 3.4.x-dev, v2.7.0, v2.7.0-BETA1, v2.7.0-BETA2, v2.7.1, v2.7.10, v2.7.11, v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.20, v2.7.21, v2.7.22, v2.7.23, v2.7.24, v2.7.25, v2.7.26, v2.7.27, v2.7.28, v2.7.29, v2.7.3, v2.7.30, v2.7.31, v2.7.32, v2.7.33, v2.7.34, v2.7.35, v2.7.4, v2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.0-BETA1, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.13, v2.8.14, v2.8.15, v2.8.16, v2.8.17, v2.8.18, v2.8.19, v2.8.2, v2.8.20, v2.8.21, v2.8.22, v2.8.23, v2.8.24, v2.8.25, v2.8.26, v2.8.27, v2.8.28, v2.8.3, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9, v3.0.0, v3.0.0-BETA1, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.0.8, v3.0.9, v3.1.0, v3.1.0-BETA1, v3.1.0-RC1, v3.1.1, v3.1.10, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7, v3.1.8, v3.1.9, v3.2.0, v3.2.0-BETA1, v3.2.0-RC1, v3.2.0-RC2, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9, v3.3.0, v3.3.0-BETA1, v3.3.0-RC1, v3.3.1, v3.3.10, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9, v3.4.0-BETA1, v3.4.0-BETA2] but these conflict with your requirements or minimum-stability.
```

Is this because the "^" is missing from the "|4.0" at framework-bundle and dependency-injection?

**Comment by bocharsky-bw** [comment]
Ah, most probably so! I made this change in https://github.com/knpuniversity/oauth2-client-bundle/commit/3df474b1b84c956e989b745589df1e912288f128 - let's wait for tests: https://travis-ci.org/knpuniversity/oauth2-client-bundle/builds/296678839

**Comment by bocharsky-bw** [comment]
FYI, now I was able to install it with:
`composer require knpuniversity/oauth2-client-bundle:dev-master`

But looks like some tests are broken :/

**Comment by weaverryan** [comment]
Yep, I was too fast and careless on that PR!  Let’s see if we can fix the tests and re-release

**Comment by weaverryan** [comment]
Let's try this again! #68

**Comment by weaverryan** [comment]
I just tagged the fix - 1.14.0. The tests now *do* pass on Symfony4. If you find any bugs now, let us know! 

Thanks!

**Comment by bocharsky-bw** [comment]
Yay, `1.14.0` works for me with Symfony 4!
